specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+    };
+
+    // The Result for a Notification Progress related operation
+    enum AppNotificationProgressResult
+    {
+        Succeeded, // The progress operation succeeded
+        AppNotificationNotFound, // The progress operation failed to find a Notification to process updates
+    };
+
+    // The Priority of the Notification UI associated with it's popup in the Action Centre
+    enum AppNotificationPriority
+    {
+        Default, // The notification should have default behavior in terms of delivery and display priority during connected standby mode.
+        High, // The notification should be treated as high priority. For desktop PCs, this means during connected standby mode the incoming notification can turn on the screen for Surface-like devices if it doesn't have a closed lid detected.
+    };
+
+    runtimeclass AppNotification
+    {
+        AppNotification(Windows.Data.Xml.Dom.XmlDocument payload);
+
+        // Unique identifier used to replace a notification within a group.
+        String Tag;
+
+        // Unique identifier for a Notification group in the app
+        String Group;
+
+        // A unique identifier for the Notification generated by the platform.
+        UInt32 Id { get; };
+
+        // The notification Xml Payload
+        Windows.Data.Xml.Dom.XmlDocument Payload{ get; };
+
+        // Gets or sets additional information about the Notification progress.
+        AppNotificationProgressData Progress;
+
+        // Gets or sets the time after which a Notification should not be displayed.
+        Windows.Foundation.DateTime Expiration;
+
+        // Indicates whether the Notification will remain in the Action Center after a reboot.
+        Boolean ExpiresOnReboot;
+
+        // Gets or sets the priority for a Notification.
+        // Hints on how and at what urgency level a notification should be presented to the user (whether to wake up the screen, etc).
+        AppNotificationPriority Priority;
+
+        // Gets or sets whether a Notification's pop-up UI is displayed on the user's screen.
+        Boolean SuppressDisplay;
+    }
+
+    // The manager class which encompasses all App Notification API Functionality
+    [experimental]
+    runtimeclass AppNotificationManager
+    {
+        // Gets a Default instance of a AppNotificationManager
+        static AppNotificationManager Default{ get; };
+
+        // Register an activator using an ActivationInfo context and caches the token for unregister
+        void RegisterActivator(AppNotificationActivationInfo details);
+
+        // Unregisters the activator and removes the cached Registration token.
+        void UnregisterActivator();
+
+        // Event handler for Notification Activations
+        event Windows.Foundation.TypedEventHandler<Microsoft.Windows.AppLifecycle.ExtendedActivationKind, AppNotificationActivatedEventArgs> AppNotificationActivated;
```

Comment:
>AppNotificationActivated [](http://example.com/codeflow?start=141&length=24)

What thread is this event raised on? Is it raised on the thread that called RegisterActivator? Is it it raised on the apartment that called RegisterActivator? Is it raised on a background thread? Is it raised on an arbitrary thread?

Can two of these events be outstanding simultaneously? Or do we serialize them?

Can this event be raised at the same time the app is dealing with an AppNotification extended activation?

Reply:
DISCUSSION: Incoming RPC thread = arbitrary background thread.

Two events can be active simultaneously.